File: replica-db/src/main/java/com/replicadb/controller/ApiController.java
package com.replicadb.controller;


import com.replicadb.service.ActorService;
import com.replicadb.service.ApiService;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

@RestController
@RequestMapping(value = "/api")
public class ApiController {
    @Autowired
    private ApiService apiService;

    @Autowired
    private ActorService actorService;

    @GetMapping(value = "/getSlave")
    public <T> T getSlave() {
        return apiService.getSlave();
    }

    @GetMapping(value = "/saveMaster")
    public void saveMaster() throws Exception {
        apiService.saveMaster();
    }

}

File: mysql/src/main/kotlin/com/mysql/controller/ApiController.kt
package com.mysql.controller

import com.fasterxml.jackson.databind.ObjectMapper
import com.mysql.service.ApiService
import org.slf4j.LoggerFactory
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.beans.factory.annotation.Qualifier
import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.RestController

@RestController
@RequestMapping(value = ["/api"])
class ApiController {
    @Autowired
    lateinit var apiService: ApiService

    private val LOGGER = LoggerFactory.getLogger(this.javaClass)

    @Autowired
    @Qualifier("customObjectMapper")
    lateinit var mapper: ObjectMapper

    @GetMapping(value = ["/testSave"])
    fun testSave() {
        apiService.testSave()
    }

    @GetMapping("/handleData")
    fun handleData(): List<*> {
        return apiService.handleData()
    }

}